### PR TITLE
firmware: Enable oversample in ADC enable instead of config

### PR DESCRIPTION
**Pull request by ben-brewer-codethink**
Oversampling registers were being set when ADC was disabled and so didn't take effect, this patch handles config and enabling properly and is tested as working.

### CODE REVIEW

**Review by tlsa — APPROVED** [review]
Looks good!

File: src/acq/adc.c [hunk 1]
```diff
@@ -483,21 +489,16 @@ enum bl_error bl_acq_adc_configure(bl_acq_adc_t *adc,
...
+		return BL_ERROR_OUT_OF_RANGE;
+	}
 
-		ADC_CFGR2(adc->base) = cfgr2;
-	} else if (shift > 0) {
-		return BL_ERROR_HARDWARE_CONFLICT;
-	} else {
-		ADC_CFGR2(adc->base) &= ~ADC_CFGR2_ROVSE;
+	if (shift > 8) {
+		return BL_ERROR_OUT_OF_RANGE;
```
Comment (kplus) [inline src/acq/adc.c:501]: a bit picky, but if return the same out of range error, why not combine them together?

**Comment by kplus** [comment]
looks good to me

File: src/acq/adc.c (hunk 1 above)
Comment (ben-brewer-codethink) [reply src/acq/adc.c:501]: We could separate them in the future, so I treat them as different errors, but I didn't think they were important enough to have their own error codes.

**Pull request merged**